src/model/document/document.ts

```diff
@@ -5,12 +5,25 @@ import { Vector2 } from '../vectors'
 import { readonly, shallowReactive, shallowRef, ref, watch } from 'vue'
 import arrayUtils from '../array-utils'
 import { ScopeElement, ScopeElementType } from './elements/scope-element'
+import { ExpressionElement, ExpressionElementType } from './elements/expression-element'
+
+type JsonType = null | boolean | number | string | JsonType[] | Vector2 | { [prop: string]: JsonType }
```

Comment:
I think we can remove `Vector2` from here again. 
`Vector2` is a class and needs some special serialization logic, which you've already written

Reply:
I agree, Vector2 does not fit in there well. During deserialization, a Vector2 is created though.

Reply:
I see, makes sense then.